PowerUp/src/Commands/AutonomousDriveWithEncoders.cpp

```diff
@@ -5,42 +5,53 @@ AutonomousDriveWithEncoders::AutonomousDriveWithEncoders(double metersToDrive, d
 
 //	initial = (Robot::driveTrain->GetPositionLeft() + Robot::driveTrain->GetPositionRight()) / 2;
 	initial = Robot::driveTrain->GetPositionRight();
+
 	distance = metersToDrive;
-	speed = speedToDrive;
+
+	if (distance > 0)
+		speed = std::abs(speedToDrive);
+	if (distance < 0)
+		speed = std::abs(speedToDrive) * -1;
+	else
+		speed = 0;
 }
 
 // Called just before this Command runs the first time
 void AutonomousDriveWithEncoders::Initialize() {
 	Robot::driveTrain->SetBrake();
+
 //	initial = (Robot::driveTrain->GetPositionLeft() + Robot::driveTrain->GetPositionRight()) / 2;
 	initial = Robot::driveTrain->GetPositionRight();
+
 	distance = round(distance * Robot::driveTrain->nativeUnitsPerMeterLowGear + initial);
 
+	RobotMap::ahrs->Reset();
 	RobotMap::ahrs->ResetDisplacement();
 }
 
 // Called repeatedly when this Command is scheduled to run
 void AutonomousDriveWithEncoders::Execute() {
+	DriverStation::ReportWarning("distance: " +std::to_string(Robot::driveTrain->GetPositionRight()) +" speed: " +std::to_string(speed));
+
 	Robot::driveTrain->ArcadeDrive(0, speed, false);
 //	Robot::driveTrain->ArcadeDrive(RobotMap::ahrs->GetAngle() * -0.03, speed, false);
 }
 
 // Make this return true when this Command no longer needs to run execute()
 bool AutonomousDriveWithEncoders::IsFinished() {
-	if (speed > 0)
-		return (Robot::driveTrain->GetPositionRight() >= distance);
-	else if (speed < 0)
-		return (Robot::driveTrain->GetPositionRight() <= distance);
-	else
-		return true;
+	return std::abs(Robot::driveTrain->GetPositionRight()) >= std::abs(distance) || speed == 0;
 }
 
 // Called once after isFinished returns true
 void AutonomousDriveWithEncoders::End() {
-	Robot::driveTrain->SetCoast();
-	Robot::driveTrain->ArcadeDrive(0, 0, true);
+	RobotMap::ahrs->Reset();
 	RobotMap::ahrs->ResetDisplacement();
-//	Wait(2);
+
+	Robot::driveTrain->ArcadeDrive(0, 0, false);
+
+	// maybe wait to give it time to brake before coasting again
+	Wait(50);
```

Comment:
What does the `Wait(50)` do? I don't think we want to pause the main thread

Reply:
I think I made a bad merge and this is why I could not drive with encoders yesterday!